inc/az_ulib_ipc_api.h

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_IPC_H
+#define AZ_ULIB_IPC_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "az_ulib_action_api.h"
+#include "az_ulib_base.h"
+#include "az_ulib_descriptor_api.h"
+#include "internal/az_ulib_ipc.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+
+#ifndef __cplusplus
+#include <stdint.h>
+#else
+#include <cstdint>
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @file    az_ulib_ipc_api.h
+ *
+ * @brief   The inter-process communication.
+ *
+ * The IPC is the component responsible to expose interfaces created by one component to all other
+ * components in the system.
+ */
+
+/**
+ * @brief IPC handle.
+ */
+typedef struct az_ulib_ipc_tag {
+  _az_ulib_ipc az_private;
+} az_ulib_ipc;
+
+/**
+ * @brief Interface handle.
+ */
+typedef _az_ulib_ipc_interface_handle az_ulib_ipc_interface_handle;
+
+/**
+ * @brief   Initialize the IPC system.
+ *
+ * This API initialize the IPC. It shall be called only once, at the beginning of the code
+ * execution.
+ *
+ * @note    This API **is not** thread safe, the other IPC API shall only be called after the
+ * initialization process is completely done.
+ *
+ * @param[in]   ipc_handle      The #az_ulib_ipc* that points to a memory position where
+ *                              the IPC shall create its control block. It cannot be `NULL`.
+ *
+ * @return The #AZ_ULIB_RESULT with the result of the initialization.
+ *  @retval #AZ_ULIB_SUCCESS                    If the IPC initialize with success.
+ *  @retval #AZ_ULIB_ILLEGAL_ARGUMENT_ERROR     If one of the arguments is invalid.
+ *  @retval #AZ_ULIB_ALREADY_INITIALIZED_ERROR  If the IPC is already initialized.
+ */
+static inline AZ_ULIB_RESULT az_ulib_ipc_init(az_ulib_ipc *ipc_handle) {
+#ifdef AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT
+  return _az_ulib_ipc_init((_az_ulib_ipc *)ipc_handle);
+#else
+  return _az_ulib_ipc_init_no_test((_az_ulib_ipc *)ipc_handle);
```

Comment:
>no_test [](start = 27, length = 7)

Nit: I wonder if we need a better word then "no_test". Maybe "no_contract" or "no_contract_check"? No test almost sounds like we won't have test cases for it.